classes/auth.php

```diff
@@ -487,6 +487,17 @@ public function should_login_redirect() {
             return false;
         }
 
+        if ($this->config->duallogin == saml2_settings::OPTION_DUAL_LOGIN_TEST && $saml == 0) {
```

Comment:
```suggestion
        if ($this->config->duallogin === saml2_settings::OPTION_DUAL_LOGIN_TEST && $saml === 0) {
```
Would this be better? (Not 100% sure about the $saml one)